engine/runtime/src/test/scala/org/enso/interpreter/test/instrument/RuntimeServerTest.scala

```diff
@@ -1324,7 +1324,7 @@ class RuntimeServerTest
           Vector(
             Api.ExpressionValueUpdate(
               idMainA,
-              Some("Number"),
+              None,
```

Comment:
Why do these become `None`? Does IDE not want to get the type here? Doesn't sound good

Reply:
If these fields become independent, it's probably better to split the message in two? The semantics of `there's a method call but no type` are unclear to me.

Reply:
There's two optional fields for the _type_ and _method pointer_ in the `ExpressionValueUpdate` message because IDE wants to see the updates only when the stuff has changed. In this case, we don't send the type because only the method pointer has changed.

PR just updates the notification logic to be more accurate, i.e. previously we always send the type whenever the method pointer changed, and now we don't)

Reply:
I see. If that's what the IDE expects really, land it :)

Reply:
Hold a second. As I know (and I asked @mwu-tow ) as well, we don't want to None here.

Reply:
Hold a second. As I know (and I asked @mwu-tow ) as well, we don't want to None here.

Reply:
The issue was that we don't want to receive update **packet** when there was no change. But if the update is fired it should contain all the information.

We treat "None" here as "There is no known type yet", no "The type was not changed from last update".

Reply:
@4e6 IDE doesn't want spurious updates, but it was about not emitting the same updates multiple times, as expression is re-evaluated without change in the code or execution context. 

Emitting an update with type `None` would actually mean "we don't know what the type is anymore" and is the desired behavior.

If we wanted to really optimize for a case where type remains the same but the method pointer changes, we should introduce separate notification for them. (or have `type` field become a 3-variant enum: `Unchanged`, `Unknown`, `Updated(typename)` — so we could handle such partial updates) — but both of these feel like an overkill.

Basically we want to have as few `ExpressionValueUpdate` as possible but not less (i.e. have them after changing frame or its properties changed). We never requested "partial" updates.

Sorry if I haven't made this clear properly before.